### PR TITLE
Add --status argument for Orders

### PR DESCRIPTION
**Pull request by shiki**
## Why

I want to use this generator in a cron job to periodically create orders for a test site. And I want to be able to receive push notifications on the [WooCommerce iOS app](https://github.com/woocommerce/woocommerce-ios) every time an order is created. I'm mainly using this for dog-fooding the iOS app. 

The problem is that, the iOS app will only receive push notifications if the Order's status is `processing`. And this generator randomizes the order status so it can be tricky to rely on. It'd be great if I can specify what status to use for the generated orders. 

## Changes

This adds a `--status` argument to generate orders with a specific status. For example, running this command would generate orders with the `processing` status:

```
wp wc generate orders 10 --status=processing
```

### Error Handling

If the user enters an invalid status, the order will be set to `pending` instead. This is done automatically by `WC_Order.set_status()`. 

## Testing

1. Run this command:

    ```
    wp wc generate orders 10 --status=completed
    ```
2. Confirm that 10 orders were created with the “Completed” status. 
3. Run this command:

    ```
    wp wc generate orders 10 
    ```
4. Confirm that 10 orders were created with random statuses.

### CODE REVIEW

File: includes/Generator/Order.php [hunk 1]
```diff
@@ -129,6 +124,26 @@ protected static function get_date_created( $assoc_args ) {
...
+	 * return a random status.
+	 *
+	 * @param array $assoc_args CLI arguments.
+	 * @return string An order status.
+	 */
+	private static function get_status( $assoc_args ) {
+		if ( ! empty( $assoc_args['status'] ) ) {
+			return $assoc_args['status'];
+		} else {
+			return self::random_weighted_element(array(
```
Comment (rrennick) [inline includes/Generator/Order.php]: Missing the whitespace before `array(`

**Review by rrennick — CHANGES_REQUESTED** [review]
@shiki Thanks for submitting the PR. This is a nice improvement and works in its current form. However, if passed `--status=test` the orders are created with a `pending` status.

Could you add a validation of the status against `wc_get_order_statuses()` and log a message if the passed status is not valid? And, if not valid, return without creating any orders.

File: includes/Generator/Order.php (hunk 1 above)
Comment (shiki) [reply includes/Generator/Order.php]: Thanks! I'm still getting used to the WP/WC conventions. 😅 

Fixed in 99e58edc7ad9cf50dce01a1ceb1868224667ebdc.

**Review by shiki — COMMENTED** [review]
Thank you very much for reviewing this, @rrennick! Sorry, it took me a while to get back to this. This is now ready for another round. 🙂 

> Could you add a validation of the status against wc_get_order_statuses() and log a message if the passed status is not valid? And, if not valid, return without creating any orders.

Good point! I added a validation in a54791662211030122c986561b9a79b7ac0781d8. I ended up using `wc_is_order_status`. How does that look?

**Review by rrennick — APPROVED** [review]
@shiki This works great. Thanks for the help.

**Pull request merged**